### PR TITLE
Add stat_activity_oldest_backend_xmin

### PR DESCRIPTION
**Pull request by benwh**
This is an experiment to see whether this is a useful metric to have: in
theory if we see the value of this gauge remain static for a large
period of time then we know that there is something running that is
preventing the xmin horizon from increasing.

### CODE REVIEW

File: collector/stat_activity.go [hunk 1]
```diff
@@ -157,5 +169,13 @@ func (c *statActivityScraper) Scrape(ctx context.Context, conn *pgx.Conn, versio
 	// postgres_stat_activity_oldest_snapshot_seconds
 	ch <- prometheus.MustNewConstMetric(c.snapshot, prometheus.GaugeValue, oldestSnapshot)
 
+	err = conn.QueryRow(ctx, statActivityScraperOldestSnapshotXidQuery).Scan(&oldestXmin)
+	if err != nil {
+		return err
+	}
+
+	// postgres_stat_activity_oldest_backend_xmin
+	ch <- prometheus.MustNewConstMetric(c.xmin, prometheus.GaugeValue, oldestXmin)
```
Comment (benwh) [inline collector/stat_activity.go]: I'm pretty sure now that this should be a counter, not a gauge (as long as my assertion about the query that retrieves this metric always holding a snapshot is correct)

Thoughts welcome though

**Comment by benwh** [comment]
@lawrencejones WDYT of this?

Have been running it in staging for the past week, although we don't see any of the flat spots there that I think we might in production.

**Comment by benwh** [comment]
Now rebased.

ping @rnaveiras

**Comment by rnaveiras** [comment]
@benwh Thanks!!

**Pull request merged**